src/main/java/net/neoforged/neoforge/event/BlockEntityTypeAddBlocksEvent.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.SignBlock;
+import net.minecraft.world.level.block.StandingSignBlock;
+import net.minecraft.world.level.block.WallSignBlock;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.neoforged.bus.api.Event;
+import net.neoforged.fml.ModLoader;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.Nullable;
+
+public class BlockEntityTypeAddBlocksEvent extends Event implements IModBusEvent {
```

Comment:
Please add some documentation to this class.

Reply:
Maybe a snippet example too. And I'd move the repetitive inheritance notes to class level.